Source/Core/Common/bbaclient.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QObject>
+
+class QTcpSocket;
+
+class BbaServer;
+class BbaDebug;
+
+class BbaClient : public QObject
+{
+  Q_OBJECT
+
+public:
+  explicit BbaClient(QTcpSocket &socket, BbaServer &server);
+
+  BbaDebug logInfo();
+
+Q_SIGNALS:
+  void receivedMessage(const QByteArray &buffer);
+
+public Q_SLOTS:
+  void sendMessage(const QByteArray &buffer);
+
+private Q_SLOTS:
+  void readyRead();
+
+private:
+  QTcpSocket &m_socket;
+  BbaServer &m_server;
+
+  QByteArray m_buffer;
+  enum { StateSize, StatePayload } m_state;
```

Comment:
This should be an enum class. Also please keep type declarations separate from their actual usages. It makes refactoring slightly less annoying if things need to be shuffled around.